### ISSUE TITLE
DataGrid aria-required-children

### ISSUE THREAD

**Issue by migsbejagh** [mui/mui-x#14273]
### Steps to reproduce

Link to live example: (required)

Steps:
1.Having a data grid component with data in it
2.Add extension Accessibility Insights for Web and run the FastPass option
3.It will have failed instance aria-required-children: Ensures elements with an ARIA role that require child roles contain them

Path | .MuiDataGrid-root

<div class="MuiDataGrid-root MuiDataGrid-root--densityStandard MuiDataGrid-withBorderColor css-bcev68-MuiDataGrid-root" role="grid" aria-colcount="9" aria-rowcount="35" aria-multiselectable="false">
#\:rk\:
#\:ro\:
#\:ru\:
#\:r12\:
button[aria-label="Go to previous page"]
button[aria-label="Go to next page"]

Fix the following:Element has children which are not allowed: button[aria-haspopup], [role=combobox], button[aria-label]

![image](https://github.com/user-attachments/assets/0f04c37f-3520-46ee-8ae3-afaa42cfc5d2)

### Current behavior

Failed instances:
aria-required-children: Ensures elements with an ARIA role that require child roles contain them

### Expected behavior

Without failed instances

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
Edge

  Output from `npx @mui/envinfo` goes here.
![image](https://github.com/user-attachments/assets/3f1d9143-ee89-4b79-a163-fc724153d6b4)

```
</details>

**Search keywords**: DataGrid; Accessibility

**Comment by migsbejagh** [comment]
it seems on 7.13 works fine
